File: .claude/scripts/hook-sprite-sync.js
#!/usr/bin/env node
/**
 * [HOOK] PostToolUse - 贴图资产三文件同步提醒
 *
 * 触发条件: 编辑 sprites.ts / sprite-viewer.html / sprite-inventory.md 任一文件
 * 输出: AUTO-AGENT: 贴图资产修改检测 → 触发 Agent(sprite-audit) 调度
 */
const fs = require('fs');

try {
  const data = JSON.parse(fs.readFileSync(0, 'utf-8'));
  const fp = data?.tool_input?.file_path || '';

  const patterns = [/sprites\.ts$/, /sprite-viewer\.html$/, /sprite-inventory\.md$/];

  if (patterns.some(p => p.test(fp))) {
    // 输出 Claude Code 自动化调度关键词（必须精确匹配 CLAUDE.md 规则）
    console.log('🤖 AUTO-AGENT: 贴图资产修改检测');

    // 详细信息供人工审查
    console.log('[HOOK] 贴图资产铁律已触发，文件:', fp);
    console.log('[HOOK] 三文件: sprites.ts + sprite-inventory.md + sprite-viewer.html');
    console.log('[HOOK] 动作: 调度 Agent(sprite-audit) 进行三文件一致性审计');
  }
} catch (e) {
  // silently fail
}

File: .claude/scripts/hook-sprite-sync.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""[HOOK] PostToolUse - 贴图资产三文件同步提醒"""
import sys, json, re, io

sys.stdout = io.TextIOWrapper(sys.stdout.buffer, encoding='utf-8', errors='replace')

try:
    raw = sys.stdin.buffer.read().decode('utf-8', errors='replace')
    data = json.loads(raw) if raw.strip() else {}
except:
    sys.exit(0)

fp = data.get("tool_input", {}).get("file_path", "")
PATTERNS = [r"sprites\.ts$", r"sprite-viewer\.html$", r"sprite-inventory\.md$"]

if any(re.search(p, fp) for p in PATTERNS):
    W = 54
    print()
    print("+" + "=" * W + "+")
    print("| [HOOK] 贴图资产铁律触发" + " " * (W - 26) + "|")
    print("+" + "-" * W + "+")
    lines = [
        f"触发文件: {fp}",
        "规则: 三文件必须同步更新",
        "必须: sprites.ts + inventory.md + viewer",
        "动作: 自动调度 Agent(sprite-audit)",
    ]
    for line in lines:
        pad = max(W - len(line.encode('utf-8')) - 1, 1)
        # 中文字符占2列宽，需要按显示宽度计算
        display_len = sum(2 if ord(c) > 127 else 1 for c in line)
        pad = max(W - display_len - 1, 1)
        print("| " + line + " " * pad + "|")
    print("+" + "=" * W + "+")
    print()
